### ISSUE TITLE
Emacs with native compilation is much slower than without (on MacOS)

### ISSUE THREAD

**Issue by monkeyjunglejuice** [doomemacs/doomemacs#8207]
### I confirm that...

- [x] I have searched the [issue tracker](https://github.com/doomemacs/doomemacs/issues), [documentation](https://docs.doomemacs.org), [FAQ](https://docs.doomemacs.org/-/faq), [Discourse](https://discourse.doomemacs.org), and [Google](https://google.com), in case this issue has already been reported/resolved.

- [x] I have read "[How to Debug Issues](https://doomemacs.org/d/how2debug)", and will use it to provide as much information about this issue as possible.

- [x] The issue can be reproduced on the **latest** available commit of Doom.

- [x] The issue can be reproduced on a stable release of Emacs, such as 27, 28, or 29. *(Unstable versions end in .50, .60, or .9x)*

### Expected behavior

Doom Emacs on Emacs 29 --with-native-comp starts faster and reacts faster than without native compilation

### Current behavior

Emacs 29.4 --with-native-comp is two times slower than without native compilation. It takes twice the time to start, and various commands lag heavily -- so much I'd call it "unusable".

### Steps to reproduce

1. Uninstall your Emacs
2. Install Emacs Plus from Homebrew:
```brew install d12frosted/emacs-plus/emacs-plus@29 --with-native-comp```
3. Start Emacs from the command line. Compilation sets in automatically. Wait for the compilation to complete. This may take several minutes.
4. Restart Emacs and check if the previous compilation process is really completed: no compilation log buffer `*Async-native-compile-log*` should be present anymore, no Emacs compilation processes in `htop -F emacs`
5. Notice if Emacs is roughly 2x slower (that's my experience); init takes longer, certain commands are lagging.
6. Uninstall it:
```brew remove d12frosted/emacs-plus/emacs-plus@29```
7. Install Emacs without native compilation (make sure that your brew install line has no `--with-native-comp` in it):
```brew install d12frosted/emacs-plus/emacs-plus@29```
8. Enjoy the double-speed (!?)

### System Information

https://pastebin.com/cpwpSvSV

  Model Name:	MacBook Pro
  Model Identifier:	MacBookPro15,1
  Processor Name:	6-Core Intel Core i7
  Processor Speed:	2.6 GHz
  Total Number of Cores:	6
  L2 Cache (per Core):	256 KB
  L3 Cache:	9 MB
  Memory:	16 GB
  System Version:	macOS 15.2 (24C101)
  Kernel Version:	Darwin 24.2.0
  Secure Virtual Memory:	Enabled
  System Integrity Protection:	Enabled

**Comment by Patryk27** [comment]
> Start Emacs. Look how slow that is.

Just in case - have you waited for the native compilation to complete? (about a couple of minutes - you can see a buffer called `*Native Compilation*` or something like that in the ibuffer when it's happening)

**Comment by monkeyjunglejuice** [comment]
Yes, I made sure of that; compilation has been completed, no compilation processes in htop. I used Emacs --with-native-comp for 3 days anyway.

**Comment by monkeyjunglejuice** [comment]
I'm currently trying to reproduce the slow Emacs again, and as far as I can tell, the lag of certain commands seems just the on-demand native compilation that is triggered when a command is run for the first time after switching to Emacs --with-native-comp.

**Comment by monkeyjunglejuice** [comment]
I consider the case as closed now as the startup- und reaction times have improved, even though there are still this weird compilation of the same files each time Emacs starts (as described here https://github.com/doomemacs/doomemacs/issues/7920)
